src/main/java/com/intel/gkl/pdhmm/IntelPDHMM.java

```diff
@@ -52,37 +54,52 @@ public boolean load(File tempDir) {
         return true;
     }
 
+    private static void checkArraySize(Object array, int expectedSize, String arrayName)
+            throws NullPointerException, IllegalArgumentException {
+        Objects.requireNonNull(array, arrayName + " must not be null.");
+        if (!array.getClass().isArray()) {
+            throw new IllegalArgumentException(arrayName + " is not an array.");
+        } else if (Array.getLength(array) != expectedSize) {
+            String errorMessage = String.format("Array %s has size %d, but expected size is %d.", arrayName,
+                    Array.getLength(
+                            array),
+                    expectedSize);
+            throw new IllegalArgumentException(errorMessage);
+        }
+    }
+
     public double[] computePDHMM(byte[] hap_bases, byte[] hap_pdbases, byte[] read_bases, byte[] read_qual,
             byte[] read_ins_qual, byte[] read_del_qual, byte[] gcp, long[] hap_lengths, long[] read_lengths,
-            int testcase, int maxHapLength, int maxReadLength)
+            int batchSize, int maxHapLength, int maxReadLength)
             throws RuntimeException, OutOfMemoryError {
```

Comment:
This "throws" clause is unnecessary. RuntimeException does not have to be declared in "throws" section. This also applies to OutOfMemoryError which is a subclass of Error which also does not need to be declared.